src/commands/classified/external.rs

```diff
@@ -48,6 +48,107 @@ impl Decoder for LinesCodec {
     }
 }
 
+fn argument_is_quoted(argument: &str) -> bool {
+    if argument.len() < 2 {
+        return false;
+    }
+
+    (argument.starts_with('"') && argument.ends_with('"')
+        || (argument.starts_with('\'') && argument.ends_with('\'')))
+}
+
+fn remove_quotes(argument: &str) -> Option<&str> {
+    if !argument_is_quoted(argument) {
+        return None;
+    }
+
+    let size = argument.len();
+
+    Some(&argument[1..size - 1])
+}
+
+fn expand_tilde<SI: ?Sized, P, HD>(input: &SI, home_dir: HD) -> std::borrow::Cow<str>
+where
+    SI: AsRef<str>,
+    P: AsRef<std::path::Path>,
+    HD: FnOnce() -> Option<P>,
+{
+    shellexpand::tilde_with_context(input, home_dir)
+}
+
+pub fn nu_value_to_string(command: &ExternalCommand, from: &Value) -> Result<String, ShellError> {
+    match &from.value {
+        UntaggedValue::Primitive(Primitive::Int(i)) => Ok(i.to_string()),
+        UntaggedValue::Primitive(Primitive::String(s))
+        | UntaggedValue::Primitive(Primitive::Line(s)) => Ok(s.clone()),
+        UntaggedValue::Primitive(Primitive::Path(p)) => Ok(p.to_string_lossy().to_string()),
+        _ => {
+            let arg = command.expect_arg("$it");
+
+            if let Some(arg) = arg {
```

Comment:
I don't understand why we're erroring in both cases here

Reply:
(shouldn't we ignore the expect and just return one error?

Reply:
Should we name it differently? see `crates/nu-parser/src/commands/classified/external.rs`

What this `expect_arg` does, is the command find in its own arguments if one is `$it` and if it finds it returns that. Returns `None` if it doesn't find it.

Reply:
I think it's own of the named arguments, rather than positional arguments